src/main/java/com/cupid/jikting/common/jwt/filter/JwtAuthenticationProcessingFilter.java

```diff
@@ -0,0 +1,94 @@
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.authority.mapping.GrantedAuthoritiesMapper;
+import org.springframework.security.core.authority.mapping.NullAuthoritiesMapper;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@RequiredArgsConstructor
+@Slf4j
+public class JwtAuthenticationProcessingFilter extends OncePerRequestFilter {
+
+    private static final String NO_CHECK_URL = "/members/login";
+
+    private final JwtService jwtService;
+    private final MemberRepository memberRepository;
+
+    private GrantedAuthoritiesMapper authoritiesMapper = new NullAuthoritiesMapper();
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request,
+                                    HttpServletResponse response,
+                                    FilterChain filterChain) throws ServletException, IOException {
+        String refreshToken = jwtService.extractRefreshToken(request).filter(jwtService::isTokenValid).orElse(null);
+
+        if (refreshToken != null) {
+            checkRefreshTokenAndReIssueAccessToken(response, refreshToken);
+            return;
+        }
+
+        if (refreshToken == null) {
+            checkAccessTokenAndAuthentication(request, response, filterChain);
+        }
+    }
+
+    public void checkRefreshTokenAndReIssueAccessToken(HttpServletResponse response, String refreshToken) {
+        memberRepository.findByRefreshToken(refreshToken).ifPresent(user -> {
+            String reIssuedRefreshToken = reIssueRefreshToken(user);
+            jwtService.sendAccessAndRefreshToken(response, jwtService.createAccessToken(user.getUsername()), reIssuedRefreshToken);
+        });
+    }
+
+    public void checkAccessTokenAndAuthentication(HttpServletRequest request,
+                                                  HttpServletResponse response,
+                                                  FilterChain filterChain) throws ServletException, IOException {
+        log.info("checkAccessTokenAndAuthentication() 호출");
+        jwtService.extractAccessToken(request).filter(jwtService::isTokenValid)
+                .flatMap(accessToken -> jwtService.extractUsername(accessToken)
+                        .flatMap(memberRepository::findByUsername))
+                .ifPresent(this::saveAuthentication);
+
+        filterChain.doFilter(request, response);
```

Comment:
p1. 지금 코드는 눈에 잘 들어오지 않는 것 같아요. flapMap을 두 번 사용하면 가독성이 더 좋아질 것 같아요!

p2. 저는 현재 메소드의 경우 빈 줄을 삽입하지 않아도 괜찮을 것 같아요. 빈 줄을 너무 많이 삽입하면 스크롤을 많이 내려야해서 가독성이 떨어지는 것 같아요.

```suggestion
        jwtService.extractAccessToken(request)
                .filter(jwtService::isTokenValid)
                .flatMap(jwtService::extractUsername)
                .flatMap(memberRepository::findByUsername)
                .ifPresent(this::saveAuthentication);
        filterChain.doFilter(request, response);
```